packages/connect/src/device/checkFirmwareRevision.ts

```diff
@@ -87,7 +95,7 @@ export const checkFirmwareRevision = async ({
 
             return { success: true };
         } catch (e) {
-            if (e.name === 'FetchError' && e.code === 'ENOTFOUND') {
```

Comment:
This catches only when network adapter is on, but there is no response or 4xx or 5xx response from server.
This did not catch error when network adapter is off. And that's is not an edge case - laptop/mobile users can routinely toggle Wi-Fi to save battery power...
Handling it in `getOnlineReleaseMetadata` makes sure we catch _any_ fetch error, while still distinguishing from `other-error` in main function body.